packages/@lwc/engine/src/framework/api.ts

```diff
@@ -532,14 +532,14 @@ export function f(items: any[]): any[] {
 }
 
 // [t]ext node
-export function t(text: string): VText {
+export function t(value: unknown): VText {
     const data = EmptyObject;
     let sel, children, key, elm;
     return {
         sel,
         data,
         children,
-        text,
+        text: String(value),
```

Comment:
this is going to slow things down for everyone, specially IE11.

Additionally, this is only one instance on which this issue is exhibit, `<div title={myArray}>` suffers from the same issue.